### PR TITLE
Ads: Summer Ads Updates

### PR DESCRIPTION
**Pull request by dbspringer**
Porting omnibus WordAds updates to Jetpack. Notably this update includes:

- **Cleanup & Optimizations**
- **2nd Belowpost Ad Option**
screen shot 2017-08-23 at 1 04 06 pm
screen shot 2017-08-23 at 1 04 22 pm

- **Criteo Adblock Acceptable Ads fallback**
screen shot 2017-08-23 at 1 06 30 pm

- **Granular Display Controls**
screen shot 2017-08-23 at 1 07 43 pm

**To Test**
1. Enable Jetpack Ads module
1. Enable `Display second ad below post`
    1. Check to see that 2nd unit appears below post
1. Enable options under `Enable below post ads on`
    1. Check below post ads do/don't display under appropriate type of page
1. Enable AdBlock Plus extension (😭)
    1. Check `Allow some non-intrusive advertising` in `Adblock Plus Options`
    1. Check that some non-obtrusive ads are in the regular spots.
screen shot 2017-08-23 at 1 18 40 pm

### CODE REVIEW

**Comment by dbspringer** [comment]
@MichaelArestad @rickybanister I could use your input on the `Enable below post ads on` copy/design.

screen shot 2017-08-23 at 1 07 43 pm

**Comment by egill** [comment]
Would it be easier to understand if we had options e.g.:
Show ads:
Front page
Posts
Pages
Archives

If a user clicks "posts", it automatically means they'll show below each post.
If posts/pages is selected, we present an option to specify 1 or 2 ad units below the ad.

File: _inc/client/traffic/ads.jsx [hunk 1]
```diff
@@ -27,7 +27,12 @@ export const Ads = moduleSettingsForm(
 		 */
 		getInitialState() {
 			return {
-				enable_header_ad: this.props.getOptionValue( 'enable_header_ad', 'wordads' )
+				enable_header_ad: 			this.props.getOptionValue( 'enable_header_ad', 'wordads' ),
```
Comment (dereksmart) [inline _inc/client/traffic/ads.jsx]: our eslint rules for [key-spacing](https://eslint.org/docs/rules/key-spacing) actually prefers that there's only one space after each colon.  No need to keep these aligned.

**Comment by dereksmart** [comment]
Left a couple minor comments, but otherwise the code and function look good!  Definitely would like the go-ahead from a designer before shipping.

**Comment by rickybanister** [comment]
I agree with @egill I think. 

It would be easier to understand if everything felt like a single set of options rather than a nested set. 

Could we do something more like:

```
[  x] Show an ad beneath each post

      *Display ads beneath posts on*
       [  x] front page
       [  x] posts
       [  x] pages
       [  x] archives

       *Additional ad placements*
       [  x] top of each page
       [  x] second ad beneath posts
```

File: modules/wordads/php/networks/amazon.php [hunk 2]
```diff
@@ -1,51 +0,0 @@
-<?php
-
-/**
- * Amazon network class
```
[the remark this answers is not in the document]
Comment (dbspringer) [reply modules/wordads/php/networks/amazon.php]: No problem 👍

**Comment by dbspringer** [comment]
@rickybanister updated the controls per your suggestions, is this what you had in mind?

screen shot 2017-08-29 at 1 58 10 pm

@dereksmart updated with your suggestions too 👍

**Comment by dereksmart** [comment]
LGTM!

**Comment by rickybanister** [comment]
Yes, that's exactly what I had in mind. I only notice one very small thing related to the copy.

You use 'below' in the first label, then 'beneath' in the second label, then 'below' again in the last option. Let's just stick with 'below' everywhere making translations easier.

**Comment by dbspringer** [comment]
Ah, good catch. Updated.

**Comment by rickybanister** [comment]
👍

**Comment by akirk** [comment]
A translator pinged me that *Display ads below posts on* … *posts* doesn't make a whole lot of sense. What about changing this to  *Display ads below content on* … *posts*?

**Comment by dbspringer** [comment]
Fine w/ me.

**Pull request merged**